### ISSUE TITLE
Protein families in Chado

### ISSUE THREAD

**Issue by oklonova** [tripal/tripal#308]
Hi,

I need some help to find a way to store information about a protein superfamily and its subfamilies in a Chado database. 
The overall structure is as follows:

>superfamily
>> family1 
>>>type1 - protein1,  protein2
>>>type2 - protein3
>>>type3 - protein4, protein5

>> family2 
>>> type4 - protein6
>>> type5 - ...
etc.

I was thinking of two ways of doing this: either 1. via a property table or 2. via hierarchy of terms in a cv (is it possible?).

1. Create new terms and use the featureprop table in a way similar to the one described here [https://sourceforge.net/p/gmod/mailman/gmod-schema/thread/1EA7B462-4D38-4628-93CD-BD3303322F29%40gmail.com/#msg36225490](url). But: how do we know that type1 belongs to family1 in this case?  

2. Create a hierarchy of new terms: one for the superfamily and several for subfamilies, where each subfamily has several types. Each protein is then associated with one type (a type is a leaf in the hierarchy). 
Is it possible to define new terms with this hierarchical structure? 

Or is there another way of representing such a classification in Chado/Tripal? 

I have read about the Group module, but did not find how I can use it. There were also some methods on this page [http://generic-model-organism-system-database.450254.n5.nabble.com/Feature-Lists-td460032.html](url), but they are not "hierarchical" in the way that is required. 

I would be grateful for any hints or examples of doing this.

**Comment by mestato** [comment]
My group is also working on storing gene family groups and/or ortholog relationships.  We looked at the group module and contacted Mara Kim - she said they have deprecated that module and they have started using dbxref as well. We are opting to go the route of using the db and cv tables. The CV would have to be created to support the hierarchy. @bradfordcondon has thought about this a good bit more than me and may be able to comment further. This could also be a good developer call topic.

**Comment by bradfordcondon** [comment]
An important question you need to ask is, how will your site display the family relationships.   The problem with using props is you won't really be granted visualizations of your families.  We are working on an entity/CV term browser mapper with Tripal 3 that might improve this but I just broke ground on it.

> But: how do we know that type1 belongs to family1 in this case?

If you went the featureprop route, you can have multiple entries for the same property with different ranks.  So at one rank you'd have type1, and the other rank you'd have family1.  Or, you'd use different featureprop terms for type and family.


>Is it possible to define new terms with this hierarchical structure?
Adding individual CVterms is limited if you're using the GUI loader- you can't define relationships etc.

However you could generate an OBO file to import.  The OBO file could include the heirarchical structure you seek.  See one of the loaded hierarchical OBO files for an example: essentially you would have terms for the families and types, and then your types would have `IS_A: [family term]`, and you'd associate your features with the types, automatically associating them with the families as well.


>Or is there another way of representing such a classification in Chado/Tripal?

As meg points out the group module is not really maintained.  The consensus seems to be its a good idea, we could use it, but a robust solution that works for all content types is clunky.

>dbxref

This provides a way to create *linkouts* for your groups.  So what some sites have done is grouped their families using *existing protein families* such as those on phytozome.  The  groupings are then added as dbxrefs allowing you to link them out to the phytozome family page.

This doesn't do much for you if your groups are de novo. (at least in terms of annotating the tree page)

Additionaly what sites have done is used the [phylotree module](https://github.com/legumeinfo/tripal_phylotree) to load in feature families as phylogenetic trees.  The features then get associated with the tree as a *proxy* for the group itself.  [Here is an example tree](https://legumeinfo.org/chado_phylotree/phytozome_10_2.59026828) and an associated [feature in that tree](https://legumeinfo.org/feature/Glycine/max/gene/Glyma.01G069400.Wm82.a2.v1) - you can see that it links out to phytozome utilizing dbxrefs.  



See also #269

**Comment by oklonova** [comment]
Thank you! Really quite a number of ways to choose from! I will try different and report later what we have adopted.

**Comment by spficklin** [comment]
Hi All,

Does it make sense to come up with a set of tables for properly storing this data?  It doesn't seem right to try and shove it into existing tables that don't really handle it well.  One suggestion was to just create group tables for each "module" (e.g. feature_group, stock_group, etc.).  It's simple and I don't think we'd have a problem getting it approved for Chado 1.4.   @bradfordcondon didn't we discuss this somewhere else? I was going to reference that discussion here but I can't find it.

**Comment by bradfordcondon** [comment]
#269   Maybe the more substantial discussion was over the phone.

I'm OK with a `*_group` schema.   presumably theres also a group table with at least a type_id, and groupprop.

**Comment by bradfordcondon** [comment]
You know, the *Relation* module i previously used could also handle grouping.  It had the problem of not being in chado because its linking entities, but I want to put it in the discussion, particularly because it also might provide a solution to the cvterm mapping relationship #297 

edit to elaborate, what the module does is you define a relationship type, and it creates a new entity type for that relationship.  relationships are themselves entities, whose job it is to define relationships between entities.  This means you get fields, and could customize what you do with displaying groups to no end.

**Comment by oklonova** [comment]
I see the ongoing discussion about the group module in Chado, but I am still curious if things could be done simply in our case.

We chose the cv route, as we have to create our own definitions of the terms anyway.
I uploaded our own cv from an .obo file which lists all the terms and relationships. Now each protein in our database is associated with a type (the type is given in the data file). This is the standard way in Chado.
 
But we also need to store information about which family each protein belongs to, display it on individual protein pages and be able to search by it (retrieve all proteins which belong to one family). Family terms are not associated with individual proteins, only with their types. Effectively, what we want is just two-level classification. This association of types and families already exits in the vocabulary.  How to "get it out"? (Tripal v2)
